uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/apis/work-pattern-api.js

```diff
@@ -13,7 +13,43 @@ define([
     return api.extend({
 
       /**
-       * This method returns the calendar for the given contact(s) and period,
+       * Assigns a work pattern to a contact
+       *
+       * @param {string} contactId
+       * @param {string} workPatternID
+       * @param {string} effectiveDate
+       * @param {string} effectiveEndDate
+       * @param {string} changeReason
+       * @param {object} params - additional parameters
+       * @return {Promise} Resolved with {Array} All Work Patterns
+       */
+      assignWorkPattern: function (contactId, workPatternID, effectiveDate, effectiveEndDate, changeReason, params) {
```

Comment:
I dont see any particular reason of not having multiple parameters, because we do the same for other similar API calls. For example https://github.com/civicrm/civihr/blob/57a3415ee63faa8632c31a1fd2028a389679d54c/uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/apis/leave-request-api.js#L91-L91

@AkA84 could you look into this?